src/contrib/cluster/Akka.DistributedData.LightningDB/LmdbDurableStore.cs

```diff
@@ -110,41 +112,24 @@ protected override void PostStop()
 
         private LightningEnvironment GetLightningEnvironment()
         {
-            LightningEnvironment env;
-
-            if (!_dirExists)
+            var t0 = Stopwatch.StartNew();
+            var env = new LightningEnvironment(_dir)
             {
-                var t0 = Stopwatch.StartNew();
-                Directory.CreateDirectory(_dir);
-                _dirExists = true;
-
-                env = new LightningEnvironment(_dir)
-                {
-                    MapSize = _mapSize,
-                    MaxDatabases = 1
-                };
-                env.Open(EnvironmentOpenFlags.NoLock);
+                MapSize = _mapSize,
+                MaxDatabases = 1
+            };
+            env.Open(EnvironmentOpenFlags.NoLock);
 
-                using (var tx = env.BeginTransaction())
-                using (tx.OpenDatabase(DatabaseName, new DatabaseConfiguration { Flags = DatabaseOpenFlags.Create }))
-                {
-                    tx.Commit();
-                }
-
-                t0.Stop();
-                if (_log.IsDebugEnabled)
-                    _log.Debug($"Init of LMDB in directory [{_dir}] took [{t0.ElapsedMilliseconds} ms]");
-            }
-            else
+            using (var tx = env.BeginTransaction())
+            using (tx.OpenDatabase(DatabaseName, new DatabaseConfiguration { Flags = DatabaseOpenFlags.Create }))
             {
-                env = new LightningEnvironment(_dir)
```

Comment:
The problem was here, the code assumes that the database is already created if the directory existed.
Just have to force LMDB to recreate the database if it doesn't exist.

Reply:
Makes sense